### DISCUSSION TITLE
nextjs intercept server side requests

### DISCUSSION THREAD

**Discussion by amirjr1378** [cypress-io/cypress#9328] [Questions and Help]
hi guys.
i want to intercept my getServerSideProps requests and mock it with my own data.
i use axios for requests.
can anyone help me to findout how ?

**Comment by osenvosem** [comment]
I'm also interested in this, how to intercept server requests.

**Comment by hect1c** [comment]
I was able to find some youtube links or articles suggesting different ways to be do this done by @bahmutov 

https://www.youtube.com/watch?v=33Hq41O0bvU
https://glebbahmutov.com/blog/mock-network-from-server/
https://www.youtube.com/watch?v=xdVRVhUUgCI&feature=youtu.be

These should be helpful. 

I was however wondering how, or what would be the best way to do this with Apollo Client and `cy.intercept` ? I've been unable to determine the best approach for this.

**Comment by Norfeldt** [comment]
@hect1c @amirjr1378 @osenvosem did you figure out a way to archive intercepting of getServerSideProps for cypress component testing?

(I added this question to stackoverflow as well https://stackoverflow.com/questions/70787370/cypress-component-testing-intercept-getserversideprops-requests)

**Comment by hello-rory** [comment]
I don't think there is a way to stub the actual server side network requests BUT you can stub the client page request and alter the response. It should look something like this:
```
export const mockServerSideProps = (pathname: string, props: Record<string, unknown>) => {
  cy.intercept("GET", `**${pathname}.json?**`, req => {
    req.reply(res => {
      res.body.pageProps = {
        ...res.body.pageProps,
        ...props,
      };
    });
  });
};
```

**Comment by JonJamesDesign** [comment]
I wrote a solution to this problem here: https://github.com/cypress-io/cypress/discussions/28469#discussioncomment-11252527

**Comment by asmyshlyaev177** [comment]
You can record/replay SSR requests, just have to pass a custom header to distinguish requests, this project grew up to a library specifically for that, will appreciate feedback - https://github.com/asmyshlyaev177/test-proxy-recorder

**Comment by jennifer-shehane** [comment]
cy.intercept() can only stub requests that pass through the browser. Next.js server-side (SSR/API-route-internal) requests never reach the browser proxy, so they can’t be intercepted. Mock at the data source instead: e.g. intercept the upstream API the server calls, or use MSW/nock on the Node side.
